src/api/HttpTransport.ts

```diff
@@ -52,21 +57,28 @@ export default class HttpTransport<Props> {
   }
 
   public request(
-    url: string,
+    path: string,
     options: HttpTransportOptions,
     timeout?: number
   ): Promise<any> {
-    const { method, data } = options;
+    const { method, data, headers } = options;
+    const url = `${BASE_API}/${path}`;
+
     return new Promise((resolve, reject) => {
       const xhr = new XMLHttpRequest();
-
+      xhr.withCredentials = true;
       if (method === METHODS.GET) {
-        xhr.open(method, url + '?' + queryStringify(data));
+        xhr.open(method, url);
       } else {
         if (typeof method === 'string') {
           xhr.open(method, url);
         }
       }
+      if (!headers) {
+        xhr.setRequestHeader('Content-Type', 'application/json');
+      } else if (headers?.contentType !== false) {
```

Comment:
В этой строке у тебя будет ошибка TS т.к. тип string сравнивается с boolean. В таком случае лучше использовать 
```suggestion
      } else if (!headers?.contentType) {
```